sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c

```diff
@@ -248,6 +248,129 @@ static int _gnrc_tcp_open(gnrc_tcp_tcb_t *tcb, char *target_addr, uint16_t targe
 }
 
 /* External GNRC TCP API */
+int gnrc_tcp_ep_init(gnrc_tcp_ep_t *ep, int family, const uint8_t *addr, size_t addr_size,
+                     uint16_t port, uint16_t netif)
+{
+#ifdef MODULE_GNRC_IPV6
+    if (family != AF_INET6) {
+        return -EAFNOSUPPORT;
+    }
+
+    if (addr == NULL && addr_size == 0) {
+        ipv6_addr_set_unspecified((ipv6_addr_t *) ep->addr.ipv6);
+    }
+    else if (addr_size == sizeof(ipv6_addr_t)) {
+        memcpy(ep->addr.ipv6, addr, sizeof(ipv6_addr_t));
+    }
+    else {
+        return -EINVAL;
+    }
+#else
+    return -EAFNOSUPPORT;
+#endif
+
+    ep->family = family;
+    ep->port = port;
+    ep->netif = netif;
+    return 0;
+}
+
+int gnrc_tcp_ep_from_str(gnrc_tcp_ep_t *ep, const char *str)
```

Comment:
Unrelated, but we should deduplicate this and the `sock_util` in sometime later on.

Reply:
I am not sure on that. Yes we have two functions that do the same, on the otherhand it keeps gnrc and sock independent from each other. I think thats up for debate.

Reply:
1. Code de-duplication of complex code is *always* a good idea
2. You can still keep the two separate by e.g. providing a common dependency ;-).